File: RukusRummy.BusinessLogic/Models/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RukusRummy.BusinessLogic.Models
{
    public class Game : Entity
    {
        public string Name { get; set; }

        public Guid Deck { get; set; }

        public List<Guid> Players { get; set; } = new List<Guid>();

        public List<Guid> Rounds { get; set; } = new List<Guid>();

        public bool AutoReveal { get; set; }
        
        public bool EnableFunFeatures { get; set; }

        public bool ShowAverage { get; set; }
        
        public bool AutoCloseSession { get; set; }
        
        public PlayerPermissionType ManageIssuesPermission { get; set; }

        public PlayerPermissionType RevealCardsPermission { get; set; }
    }
}

File: RukusRummy.Web/src/Models/Game.ts
import IDeck from "./Deck";
import { IPlayer } from "./Player";

export interface ICreateGameRequest {
    name?: string;
    playerId: string;
    deckId: string;
    revealCardsPermission?: string;
    manageIssuesPermission?: string;
    autoReveal: boolean;
    enableFunFeatures: boolean;
    showAverage: boolean;
}

export interface ICreateDeckRequest {
    name?: string;
    values?: string;
}

export enum PlayerPermissionType {
    AllPlayers,
    JustMe
}

export enum GameStateType {
    RoundActive,
    RoundFinished
}


export interface IGame {
    id: string;
    name: string;
    state: GameStateType;
    deck: IDeck;
    players: Array<IPlayer>;
    rounds: Array<IRound>;
    autoReveal: boolean;
    enableFunFeatures: boolean;
    showAverage: boolean;
    autoCloseSessionL: boolean;
    manageIssuesPermission: PlayerPermissionType;
    revealCardsPermission: PlayerPermissionType;
}

export interface IVoteTally {
    [key:string]: number | undefined
}

export interface IRound
{
    name: string;
    result: string;
    startDate: Date;
    endDate: Date;
    votes: IVoteTally;
    voteCount: number;
    average: string;
}